### PR TITLE
feat: add mamba backend

### PR DESCRIPTION
**Pull request by mederka**

### CODE REVIEW

**Review by chamini2 — APPROVED** [review]
LGTM! 🥇

**Review by turbo1912 — APPROVED** [review]

File: src/isolate/backends/conda.py [hunk 1]
```diff
@@ -39,6 +40,7 @@
 @dataclass
 class CondaEnvironment(BaseEnvironment[Path]):
     BACKEND_NAME: ClassVar[str] = "conda"
+    resolver: str = "mamba"
```
Comment (isidentical) [inline src/isolate/backends/conda.py:43]: Let's put it below the existing fields

File: src/isolate/backends/conda.py [hunk 2]
```diff
@@ -144,23 +150,33 @@ def destroy(self, connection_key: Path) -> None:
...
             )
 
+    def _run_mamba(self, *args: Any) -> None:
+        mamba_executable = _get_executable("mamba")
+        with logged_io(partial(self.log, level=LogLevel.INFO)) as (stdout, stderr):
+            subprocess.check_call(
+                [mamba_executable, *args],
+                stdout=stdout,
+                stderr=stderr,
+            )
```
Comment (isidentical) [inline src/isolate/backends/conda.py:177]: Not super sure where this command is called?

File: src/isolate/backends/conda.py [hunk 3]
```diff
@@ -144,23 +150,33 @@ def destroy(self, connection_key: Path) -> None:
...
-                "--prefix",
-                connection_key,
-            )
+            self._run_destroy(str(connection_key))
+
+    def _run_create(self, env_path: str, env_name: str) -> None:
+        self._run_conda("env", "create", "--force", "--prefix", env_path, "-f", env_name)
+
+    def _run_destroy(self, connection_key: str) -> None:
+        self._run_conda("remove","--yes","--all","--prefix", connection_key)
```
Comment (isidentical) [inline src/isolate/backends/conda.py:159]: Can we just change `_run_conda` to use the resolver by default and leave the call sites as is (if there is no change in the arg formatting~)

File: src/isolate/backends/conda.py [hunk 4]
```diff
@@ -170,9 +186,17 @@ def open_connection(self, connection_key: Path) -> PythonIPC:
...
-        conda_path = shutil.which(_CONDA_COMMAND, path=path)
+def _get_executable(exec_type: str) -> Path:
+    if exec_type == "conda":
+        home = _ISOLATE_CONDA_HOME
+        command = _CONDA_COMMAND
+    elif exec_type == "mamba":
+        home = _ISOLATE_MAMBA_HOME
+        command = _MAMBA_COMMAND
+    else:
+        raise Exception(f"Executable of type {exec_type} is not supported")
```
Comment (isidentical) [inline src/isolate/backends/conda.py:197]: This doesn't need to validate the `exec_type`, the validation should happen in the `from_config` method

File: tests/test_backends.py [hunk 5]
```diff
@@ -378,14 +378,23 @@ def test_tags_in_key(self, tmp_path, monkeypatch):
 # Since conda is an external dependency, we'll skip tests using it
 # if it is not installed.
 try:
-    _get_conda_executable()
+    _get_executable("conda")
```
Comment (isidentical) [inline tests/test_backends.py:381]: We can drop this part completely

File: pyproject.toml [hunk 6]
```diff
@@ -31,6 +31,7 @@ build = ["virtualenv", "PyYAML"]
 [tool.poetry.plugins."isolate.backends"]
 "virtualenv" = "isolate.backends.virtualenv:VirtualPythonEnvironment"
 "conda" = "isolate.backends.conda:CondaEnvironment"
+"mamba" = "isolate.backends.mamba:MambaEnvironment"
```
Comment (isidentical) [inline pyproject.toml:34]:
```suggestion
"mamba" = "isolate.backends.mamba:MambaEnvironment"
```

**Review by isidentical — APPROVED** [review]

File: src/isolate/backends/conda.py [hunk 7]
```diff
@@ -170,9 +183,9 @@ def open_connection(self, connection_key: Path) -> PythonIPC:
 
 
 @functools.lru_cache(1)
-def _get_conda_executable() -> Path:
-    for path in [_ISOLATE_CONDA_HOME, None]:
-        conda_path = shutil.which(_CONDA_COMMAND, path=path)
+def _get_executable(home: str, command: str) -> Path:
```
Comment (isidentical) [inline src/isolate/backends/conda.py:186]:
```suggestion
def _get_executable(command: str, home: str | None = None) -> Path:
```

**Review by isidentical — COMMENTED** [review]
the key property should also include resolver

File: src/isolate/backends/conda.py (hunk 3 above)
Comment (mederka) [reply src/isolate/backends/conda.py:167]: There's change in arg formatting though 😔

**Review by isidentical — APPROVED** [review]

**Pull request merged**